src/provisioner/cluster.py

```diff
@@ -65,17 +78,106 @@ def _get_kvs(self, prefix, node):
-        elif isinstance(node, str):
+        elif isinstance(node, str) or isinstance(node, int):
             kvs.append((prefix, node))
         return kvs
 
     def save(self, config_store):
         """ Saves cluster information onto the conf store """
 
         kvs = []
-        for node in self._node_list:
-            node_id = node['id']
-            key_prefix = f'node>{node_id}'
-            kvs.extend(self._get_kvs(key_prefix, node))
+        try:
+            for node in self._node_list:
+                node_id = node.pop('id')
+                components = {}
+                for component in node.pop('components'):
+                    comp_name = component.get('name')
+                    components.update({
+                        comp_name:  component.get('services')
+                        })
+                node['components'] = components
+                if node['type'] == 'storage_node':
+                    cvg_list = node.pop("storage")
+                    node['storage'] = {
+                        'cvg_count': len(cvg_list),
+                        'cvg': cvg_list
+                    }
+
+                key_prefix = f'node>{node_id}'
+                kvs.extend(self._get_kvs(key_prefix, node))
+
+            config_store.set_kvs(kvs)
+
+        except KeyError as e:
+            raise CortxProvisionerError(
+                errno.EINVAL,
+                f'Error occurred while adding node information into confstore {e}')
+
+
+class CortxStorageSet:
+    """ Represents CORTX storage_set """
+
+    def __init__(self, storage_sets: list = []):
+        self._storage_sets = storage_sets
+        for s_set in self._storage_sets:
+            CortxStorageSet._validate(s_set)
+
+    @staticmethod
+    def _validate(s_set: dict):
+        """
+        validates a give storage_sets to have required properties
+        Raises exception if there is any entry missing
+        """
+        s_set_name = s_set.get('name')
+        if s_set_name is None:
+            raise CortxProvisionerError(
+                errno.EINVAL, 'Missing name for the storage_set entry')
+        required_keys_for_storage_set = ['durability', 'nodes']
+        for k in required_keys_for_storage_set:
+            if s_set.get(k) is None:
+                raise VError(
+                    errno.EINVAL,
+                    f"'{k}' property is unspecified for storage_set {s_set_name}.")
+
+    def save(self, config_store):
+        """ Converts storage_set confstore keys
+        and add into conf_store.
+        constore representation for storage_set key:
+         storage_set:
+            - durability:
+                dix:
+                    data: '1'
+                    parity: '7'
+                    spare: '0'
+                sns:
+                    data: '8'
+                    parity: '7'
+                    spare: '0'
+                name: storage-set-1
+                nodes: []
```

Comment:
nodes will not stored in the config_store under storage_set

Reply:
Removed.